plugins/overwatch.rb

```diff
@@ -0,0 +1,660 @@
+# Overwatch plugin - Active Routing and Session Management
+#
+# Author: sn0wfa11 - jhale85446[at]gmail.com
+#
+# Session stepping and logging functionality borrowed on HDM's Beholder plugin.
+
+require 'fileutils'
+
+module Msf
+
+class Plugin::Overwatch < Msf::Plugin
+
+  # Worker Thread
+  #
+  # @return [void] A useful return value is not expected here
+
+  class OverwatchWorker
+    attr_accessor :framework, :config, :driver, :thread
+    attr_accessor :state
+
+    def initialize(framework, config, driver)
+      self.state     = { }
+      self.framework = framework
+      self.config    = config
+      self.driver    = driver
+      self.thread    = framework.threads.spawn('OverwatchWorker', false) {
+        begin
+          self.start
+        rescue ::Exception => e
+          $stderr.puts "OverwatchWorker: #{e.class} #{e.to_s} #{e.backtrace}"
+        end
+
+        # Mark this worker as dead
+        self.thread = nil
+      }
+    end
+
+    def stop
+      return unless self.thread
+      self.thread.kill rescue nil
+      self.thread = nil
+    end
+
+    # Starts the overwatch worker and loops through the open sessions
+    # based on the defined timeframe.
+    # Primary error handling to the log file occurs here as well.
+    #
+    # @return [void] A useful return value is not expected here
+    def start
+      if self.config[:log]
+        self.driver.print_status("Overwatch is logging to #{self.config[:base]}. Use 'overwatch_start -h' for plugin information.")
+      else
+        self.driver.print_status("Overwatch started, logging is disabled. Use 'overwatch_start -h' for plugin information.")
+      end
+
+      bool_options = [ :autoroute, :reroute_stale, :kill_stale, :kill_stale_dup, :log ]
+      bool_options.each do |o|
+        self.config[o] = !!( self.config[o].to_s =~ /^[yt1]/i)
```

Comment:
I used @hdm's Beholder plugin as a template. This line is directly from Beholder (line 45). I messed with it a bit when I started but it got really pissed when I changed stuff. So I took an "if it ain't broke, don't fix it" stance. I'll look at it again to see what I can figure out. HDM would definitely be better at this line than me.